src/app/components/elements/panels/EventGenderDetails.tsx

```diff
@@ -0,0 +1,106 @@
+import React from "react";
+import { EventBookingDTO, UserRole } from "../../../../IsaacApiTypes";
+import { Col, Table, UncontrolledTooltip } from "reactstrap";
+import { asPercentage } from "../../../services";
+
+export const countEventDetailsByRole = (role: UserRole | undefined, eventBookings: EventBookingDTO[]) => {
+  const countByRole = {
+    genders: {
+      male: 0,
+      female: 0,
+      other: 0,
+      preferNotToSay: 0,
+      unknown: 0,
+    },
+    numberOfConfirmedOrAttendedBookings: 0,
```

Comment:
This isn't necessarily wrong, but it's unnecessary, because this value is always going to be a sum of all the gender values. So by removing this (and `genders`), you would be simplifying the return type of this method. But, as I say, it's not wrong as it is, so don't feel like you have to change it.

Reply:
`numberOfConfirmedOrAttendedBookings` is used in the code, so if I don't return this then I'd still have to perform some kind of calculation anyway, even if that was just declaring a variable with the sum of the gender values - I think that approach might be slightly less readable?

Reply:
Yes, you're right, you'll have to declare a variable in that case, but it wouldn't be less readable, since you can still call that variable `numberOfConfirmedOrAttendedBookings`. But, as I said, the way you've done it is not wrong, so feel free to leave it like that. I mentioned it because to me this kind of normalised data makes more sense when it would be computationally expensive to calculate, but in this case it's just an addition of 5 values, which is O(1). But leave it as it is, it's fine.